collectors/disk.go

```diff
@@ -104,7 +106,7 @@ func (c *Disk) scrape() error {
 	for diskPath, usage := range dev {
 		if len(c.allowedDisks) > 0 {
 			_, diskName := path.Split(diskPath) // return "sda1" from "/dev/sda1"
-			if _, allowed := c.allowedDisks[diskName]; !allowed {
+			if !stringInSlice(diskName, c.allowedDisks) {
```

Comment:
it's just a `contains`, not a regex match ?

Reply:
Check the stringInSlice implementation ;)